File: android/app/src/main/java/com/texasgamer/zephyr/util/lifecycle/ZephyrLifecycleLogger.java
package com.texasgamer.zephyr.util.lifecycle;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import com.texasgamer.zephyr.ZephyrApplication;
import com.texasgamer.zephyr.util.log.ILogger;
import com.texasgamer.zephyr.util.log.LogLevel;

import javax.inject.Inject;

/**
 * Logs lifecycle events.
 */
public class ZephyrLifecycleLogger implements Application.ActivityLifecycleCallbacks {

    private static final String LOG_TAG = "ZephyrLifecycleLogger";

    @Inject
    ILogger logger;

    public ZephyrLifecycleLogger() {
        ZephyrApplication.getApplicationComponent().inject(this);
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityCreated: " + activity.getLocalClassName());
    }

    @Override
    public void onActivityStarted(Activity activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityStarted: " + activity.getLocalClassName());
    }

    @Override
    public void onActivityResumed(Activity activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityResumed: " + activity.getLocalClassName());
    }

    @Override
    public void onActivityPaused(Activity activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityPaused: " + activity.getLocalClassName());
    }

    @Override
    public void onActivityStopped(Activity activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityStopped: " + activity.getLocalClassName());
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivitySaveInstanceState: " + activity.getLocalClassName());
    }

    @Override
    public void onActivityDestroyed(Activity activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityDestroyed: " + activity.getLocalClassName());
    }
}

File: android/app/src/main/java/com/texasgamer/zephyr/util/lifecycle/ZephyrLifecycleLogger.kt
package com.texasgamer.zephyr.util.lifecycle

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle
import com.texasgamer.zephyr.ZephyrApplication
import com.texasgamer.zephyr.util.log.ILogger
import com.texasgamer.zephyr.util.log.LogLevel
import javax.inject.Inject

/**
 * Logs lifecycle events.
 */

private const val LOG_TAG = "ZephyrLifecycleLogger"

class ZephyrLifecycleLogger(private val logger: ILogger) : ActivityLifecycleCallbacks {

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityCreated: " + activity.localClassName)
    }

    override fun onActivityStarted(activity: Activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityStarted: " + activity.localClassName)
    }

    override fun onActivityResumed(activity: Activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityResumed: " + activity.localClassName)
    }

    override fun onActivityPaused(activity: Activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityPaused: " + activity.localClassName)
    }

    override fun onActivityStopped(activity: Activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityStopped: " + activity.localClassName)
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivitySaveInstanceState: " + activity.localClassName)
    }

    override fun onActivityDestroyed(activity: Activity) {
        logger.log(LogLevel.INFO, LOG_TAG, "onActivityDestroyed: " + activity.localClassName)
    }
}